lib/galaxy/tools/evaluation.py

```diff
@@ -540,13 +540,14 @@ def __build_environment_variables(self):
                 is_template = False
             else:
                 is_template = True
-            with tempfile.NamedTemporaryFile(dir=directory, delete=False) as temp:
+            with tempfile.NamedTemporaryFile(dir=directory, prefix="tool_env_", delete=False) as temp:
                 config_filename = temp.name
             self.__write_workdir_file(config_filename, environment_variable_template, param_dict, is_template=is_template, strip=environment_variable_def.get("strip", False))
             config_file_basename = os.path.basename(config_filename)
             # environment setup in job file template happens before `cd $working_directory`
-            environment_variable["value"] = '`cat "$_GALAXY_JOB_DIR/%s"`' % config_file_basename
+            environment_variable["value"] = '`cat "%s/%s"`' % (self.compute_environment.env_config_directory(), config_file_basename)
             environment_variable["raw"] = True
+            environment_variable["job_directory_path"] = config_filename
```

Comment:
Is this a file path, and should this be a directory path ? Maybe `self.compute_environment. working_directory()` ?

Reply:
I meant this as "path" below "job_directory". ``compute_environment.working_directory()`` doesn't work on Pulsar because it is the tool's directory not the jobs there.